### PR TITLE
feat: support bins attr in bzlmod

### PR DESCRIPTION
**Pull request by kormide**
In support of https://github.com/aspect-build/rules_js/issues/644.

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -6,11 +6,19 @@ module(
...
 bazel_dep(name = "platforms", version = "0.0.4")
 
+bazel_lib_ext = use_extension("@aspect_bazel_lib//lib:extensions.bzl", "ext")
+
+use_repo(bazel_lib_ext, "copy_to_directory_toolchains")
+
+register_toolchains(
+    "@copy_to_directory_toolchains//:all",
+)
+
```
Comment (kormide) [inline MODULE.bazel:21]: `npm_package` makes use of the toolchain.

File: e2e/pnpm_workspace_rerooted/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+npm = use_extension("@aspect_rules_js//npm:extensions.bzl", "npm")
+
+npm.npm_translate_lock(
+    name = "npm",
+    bins = {
+        # derived from "bin" attribute in node_modules/typescript/package.json
+        "typescript": [
+            "tsc=./bin/tsc",
+            "tsserver=./bin/tsserver",
+        ],
```
Comment (kormide) [inline e2e/pnpm_workspace_rerooted/MODULE.bazel:41]: Note the alternative form we have to input this under a bzlmod extension.

File: npm/npm_import.bzl [hunk 3]
```diff
@@ -355,6 +355,13 @@ WARNING: `package_json` attribute in `npm_translate_lock(name = "{name}")` is de
     if type(bins) != "dict":
         fail("Expected bins to be a dict")
     for key, value in bins.items():
+        if type(value) == "list":
+            # The passed 'bins' value is already in the dict-of-string-list
+            # form needed by the rule. This is undocumented but necessary for
+            # the bzlmod interface to use this macro since dict-of-dicts attributes
+            # cannot be passed into module extension attrs.
+            bins_string_list_dict = bins
+            break
```
Comment (kormide) [inline npm/npm_import.bzl:364]: I hope stardoc will soon generate docs for module extensions.

Comment (kormide) [reply npm/npm_import.bzl:364]: Should I document this in the macro docstring or leave undocumented for now?

File: MODULE.bazel (hunk 1 above)
Comment (gregmagolan) [reply MODULE.bazel:21]: This should unnecessary after https://github.com/aspect-build/bazel-lib/pull/336?

Comment (gregmagolan) [reply MODULE.bazel:21]: https://github.com/bazelbuild/bazel-central-registry/pull/372 will have what you need here

Comment (gregmagolan) [reply MODULE.bazel:21]: I'll land the version bump in rules_js here https://github.com/aspect-build/rules_js/pull/789

File: e2e/pnpm_workspace_rerooted/MODULE.bazel (hunk 2 above)
Comment (gregmagolan) [reply e2e/pnpm_workspace_rerooted/MODULE.bazel:41]: makes sense 👍

File: npm/npm_import.bzl (hunk 3 above)
Comment (gregmagolan) [reply npm/npm_import.bzl:370]: Lets just change the WORKSPACE version to be the same format in a follow-up and change the docs there. The current syntax can be supported as an undocumented variant using typeof.

**Review by gregmagolan — APPROVED** [review]

**Pull request merged**